### PR TITLE
Feature/serial payload example

### PR DESCRIPTION
**Pull request by evanShap**
This is the demo code for the upcoming "Bristlemouth Dev Kit Guide: Integrating an RS232 Serial Sensor" guide [internal draft](https://www.notion.so/bristlemouth/Internal-Draft-Bristlemouth-Dev-Kit-Guide-Integrating-an-RS232-Serial-Sensor-abb7d4c9dd38454f941cbec92182142b#1c3921dea2664214b314735443b5af2e).

Ideally this is merged in and released as part of 0.6.0 prior to BristleCon on November 9.

### CODE REVIEW

File: DEVELOPER.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ This document (hopefully) details the where/why/how of bristlemouth firmware. Fo
 
 If you're developing on the Bristlemouth Development Kit, see the [BMDK_README.md](./src/apps/bm_devkit/BMDK_README.md).
 
+# Running Tests
+### Build Tests
+To run build tests locally, run `python3 tools/scripts/test/did_i_break_something.py tools/scripts/test/configs`.  This script will verify all apps compile successfully.
```
Comment (victorsowa12) [inline DEVELOPER.md:9]: do we list anywhere how to do just `make test` in the cmake directory or how to do `ctest`? Might be

Comment (evanShap) [reply DEVELOPER.md:9]: not sure, let's add in separate PR if needed?

File: src/apps/bm_devkit/serial_payload_example/user_code/user_code.cpp [hunk 2]
```diff
@@ -81,8 +100,40 @@ void loop(void) {
...
+    printf("%02X ", byte_read);
+  }
+  if (readingBytesTimer > -1 &&
+      (u_int32_t)uptimeGetMs() - (u_int32_t)readingBytesTimer >= BYTES_CLUSTER_MS) {
+    printf("\n");
+    readingBytesTimer = -1;
+  }
+
   // Read a line if it is available
+  // Note - PLUART::setUseLineBuffer must be set true in setup to enable lines.
```
Comment (victorsowa12) [inline src/apps/bm_devkit/serial_payload_example/user_code/user_code.cpp:130]: what happens if you enable setUseLiineBuffer and setUserByteStreamBuffer? Do they work simultaneously?

**Review by victorsowa12 — APPROVED** [review]
Got a question about both the byte stream buffer and line reading working simultaneously, but looks good otherwise 👍

File: src/apps/bm_devkit/serial_payload_example/user_code/user_code.cpp (hunk 2 above)
Comment (victorsowa12) [reply src/apps/bm_devkit/serial_payload_example/user_code/user_code.cpp:130]: i see that they are both enabled by default, but I just want to confirm that they can work at the same time

Comment (evanShap) [reply src/apps/bm_devkit/serial_payload_example/user_code/user_code.cpp:130]: Yep - they don't stomp on each other. PLUART will process either or both depending on the configurations.

**Pull request merged**